src/Features/CSharp/Portable/CodeRefactorings/CSharpRefactoringHelpersService.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Composition;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.ExtractMethod;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings
+{
+    [ExportLanguageService(typeof(IRefactoringHelpersService), LanguageNames.CSharp), Shared]
+    internal class CSharpRefactoringHelpersService : RefactoringHelpersService
+    {
+        public override SyntaxNode ExtractNodeFromDeclarationAndAssignment<TNode>(SyntaxNode node)
+        {
+            switch (node)
+            {
+                case LocalDeclarationStatementSyntax localDeclaration:
+                    {
+                        if (localDeclaration.Declaration.Variables.Count == 1 && localDeclaration.Declaration.Variables.First().Initializer != null)
+                        {
+                            var initializer = localDeclaration.Declaration.Variables.First().Initializer;
+                            return initializer.Value as TNode;
+                        }
+
+                        break;
+                    }
```

Comment:
1. outer blocks not necessary.
2. i think these helpers should document the theory behind things.

Reply:
1. The blocks around cases? I'd keep them for two reasons a) this is how VS generated it. b) this way scoping is more local. 
2. Will add comments